pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -492,6 +493,7 @@ def _do_catch(self, pokemon, encounter_id, catch_rate_by_ball, is_vip=False):
                     data={'pokemon': pokemon.name}
                 )
                 used_berry = False
+                catch_rate_by_ball = original_catch_rate_by_ball
```

Comment:
I was literally just going to put in a PR for this. Your solution is much simpler.